src/compiler/checker.ts

```diff
@@ -21820,10 +21829,18 @@ namespace ts {
 
         function checkBinaryLikeExpression(left: Expression, operatorToken: Node, right: Expression, checkMode?: CheckMode, errorNode?: Node): Type {
             const operator = operatorToken.kind;
+            let leftType: Type;
+
+            const symbol = (<Identifier>left).escapedText ? getResolvedSymbol(left as Identifier) : undefined;

```

Comment:
Use `isEntityNameExpression` and not `left.escapedText` (since checking `a.b += 1` is also desireable. Also, a test for `a.b += 1` would be good.